tests/test_outside_project_root.py

```diff
@@ -10,20 +10,23 @@
 
 from skbuild.exceptions import SKBuildError
 
-from . import project_setup_py_test
+from . import project_setup_py_test, push_dir
 
 
 def test_outside_project_root_fails():
-    @project_setup_py_test(("samples", "fail-outside-project-root"),
-                           ["install"],
-                           clear_cache=True)
-    def should_fail():
-        pass
-
-    exception_thrown = False
-    try:
-        should_fail()
-    except SKBuildError:
-        exception_thrown = True
-
-    assert exception_thrown
+
+    with push_dir():
```

Comment:
This shouldn't be necessary, since the `project_setup_py_test` decorator should already take care of that.